### ISSUE TITLE
Puppeteer/Chromium rendering issues?

### ISSUE THREAD

**Issue by chancancode** [actions/virtual-environments#385]
I'm not sure if this is a bug, or that it is related to GitHub's virtual environments at all, but I am running out of ideas to debug this further, so here is a last-ditch effort to see if it rings a bell for anyone here.

**Describe the bug**

We have a cron job that builds the [Ember tutorial](https://guides.emberjs.com/release/tutorial/) daily ([source here](https://github.com/ember-learn/super-rentals-tutorial)). The idea is that the build will follow the steps in the tutorial the same way an end-user would (generate an app, editing the generated files, etc) and capture the command line output along the way, as well as taking screenshots of the app being built using [puppeteer](https://github.com/puppeteer/puppeteer), which uses Chromium under-the-hood.

Since around four days ago ([last good build](https://github.com/ember-learn/super-rentals-tutorial/runs/422227583?check_suite_focus=true), [first bad build](https://github.com/ember-learn/super-rentals-tutorial/runs/424348170?check_suite_focus=true)), things started to render differently in puppeteer in a subtle but noticeable way. You can [see for yourself here](https://github.com/ember-learn/guides-source/pull/1289/files), using the "Swipe" or "Onion Skin" mode to review the differences.

TL;DR fonts a rendering differently (different letter spacing, kerning, padding?):

<img width="293" alt="Screen Shot 2020-02-07 at 2 49 44 PM" src="https://user-images.githubusercontent.com/55829/74071571-25ca1700-49b9-11ea-8555-ee3bb3b4b9e7.png">

<img width="245" alt="Screen Shot 2020-02-07 at 2 48 49 PM" src="https://user-images.githubusercontent.com/55829/74071621-4d20e400-49b9-11ea-82ff-811ad6383ac4.png">

My first thought is that perhaps this is due to a Chrome upgrade on the virtual environment. However, I don't think that's the case. We install puppeteer via yarn with a lockfile. Puppeteer (unlike puppeteer-core) bundles its own version of Chromium, so I don't think we are using the system Chrome at all. And the lockfile in yarn should ensure we are getting the same version of puppeteer (and therefore Chromium) across the build.

I'm hoping to see if the timing of this happens to coincide with rolling out some kind of changes to the hosted virtual environments. Perhaps upgrading/changing fontconfig, or other system packages/pre-installed fonts ([mscorefonts](https://www.ostechnix.com/install-microsoft-windows-fonts-ubuntu-16-04/]?) that could have this kind of effect?

I should add that I don't necessarily think either of the before/after rendering result is "more correct". They are both fine for our purpose as long as they stay consistent.

**Area for Triage**: JavaScript and Node.js, Packages

**Question, Bug, or Feature?**: Question

**Virtual environments affected**
- [ ] macOS 10.15
- [ ] Ubuntu 16.04 LTS
- [x] Ubuntu 18.04 LTS
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Expected behavior**
Consistent rendering across builds.

**Actual behavior**
It's still... consistent? But the result is distinctly different before and after Feb 3rd, so _something_ has changed.

**Comment by chancancode** [comment]
Update: still not really sure what happened, but a colleague mentioned a client had the same issue with the same font, also loaded using Google fonts. I filed https://github.com/google/fonts/issues/2334 to see if a regression on their side is a possibility.

**Comment by maxim-lobanov** [comment]
@chancancode, Is this issue still actual?
I am not sure that it is something related to the image (at least, didn't see related changes) but if we can help with your investigation somehow, please let us know

**Comment by chancancode** [comment]
Yeah, it hasn’t “changed back” since. It basically changed from A to B suddenly one day and has been fairly consistent since. The image is one of the two variables I can think of right now. The other one being the online resource (Google Font). I filed the issue in both, but because I have relatively little visibility into both, I’m a bit stuck on debugging this. That being said, as long as it doesn’t happen again, for my purpose it doesn’t really matter. It’s a big deal whenever it changes (because they churn a lot of image bytes that didn’t actually change) but as long as it’s consistent, either of the visual outputs are fine for us.

**Comment by dmitry-shibanov** [comment]
Hello, thank you for your response. We have investigated this issue deeper from our side and didn't find the root cause, unfortunately.
We have tried to downgrade Chrome / ChromeDriver on the image. Also we have looked at the software updates at the time when you have reported the problem.
It looks like nothing was changed on image side related to fonts / Chrome version.
Probably, issue can come from pupiter version but unfortunately, we are not familiar with this test framework so we can't guess if it can be related.
Please let us know if we can help you somehow else

**Comment by dmitry-shibanov** [comment]
Hello, I am closing this issue for now because as I mentioned above, that doesn’t look like image issue. Feel free to ping me if you found out image issue.
